hydromt/models/model_lumped.py

```diff
@@ -122,14 +145,21 @@ def write_response_units(
         """
         nc_dict = dict()
         if len(self._response_units) > 0:
-            # write geometry
+            # write geometries
             ds = self._response_units
-            gdf = gpd.GeoDataFrame(geometry=ds["geometry"].values, crs=ds.rio.crs)
-            if not isdir(dirname(join(self.root, fn_geom))):
-                os.makedirs(dirname(join(self.root, fn_geom)))
-            gdf.to_file(join(self.root, fn_geom), driver="GeoJSON")
+            # get list of coordinates that hold geometries
+            corl = list(ds.coords)
+            geom_coords = [v for v in corl if 'geometry' in v]
```

Comment:
Nice addition to write multiple geometries! I wonder if we cannot better do a data type check rather than a name check. I.e. you could check if the first value of every 1D coordinate is an instance of a shapely BaseGeometry.

Reply:
Yes that could be a check, likely to be more fail safe (in combination with a naming convention that involves the word geometry I would say, to keep that from geoDataframes.) I put it in as a TODO recommendation for now.